frontend/src/Components/StudentMetrics.tsx

```diff
@@ -0,0 +1,197 @@
+import { useEffect, useState } from 'react';
+import useSWR from 'swr';
+import { AxiosError } from 'axios';
+import API from '@/api/api';
+import { Facility, LoginMetrics, ServerResponseOne } from '@/common';
+import {
+    BarChart,
+    CartesianGrid,
+    XAxis,
+    YAxis,
+    Tooltip,
+    Legend,
+    Bar
+} from 'recharts';
+import PrimaryButton from './PrimaryButton';
+
+const OperationalInsights = () => {
+    const [facilities, setFacilities] = useState<Facility[]>();
+    const [facility, setFacility] = useState('all');
+    const [days, setDays] = useState(7);
+    const [resetCache, setResetCache] = useState(false);
+
+    const { data, error, isLoading, mutate } = useSWR<
+        ServerResponseOne<LoginMetrics>,
+        AxiosError
+    >(
+        `/api/login-metrics?facility=${facility}&days=${days}&reset=${resetCache}`
+    );
+
+    useEffect(() => {
+        void mutate();
+    }, [facility, days, resetCache]);
+
+    useEffect(() => {
+        const fetchFacilities = async () => {
+            const response = await API.get<Facility>('facilities');
+            setFacilities(response.data as Facility[]);
+        };
+        void fetchFacilities();
+    }, []);
+
+    const metrics = data?.data;
+
+    return (
+        <div className="p-8">
+            {error && <div>Error loading data</div>}
+            {!data || (isLoading && <div>Loading...</div>)}
+            {data && metrics && (
+                <>
+                    <PrimaryButton
+                        className="w-30"
+                        onClick={() => setResetCache(!resetCache)}
+                    >
+                        Refresh Data
+                    </PrimaryButton>
+                    <div className="grid gap-4 mb-6">
+                        <div className="stats shadow mb-6">
```

Comment:
I dont love the use of stat here, especially since we are trying to move away from daisy and more into using our own classes. there is a class called "card" which should have a correct layout (i think) similar to these, but puts them into their own individual cards. "grid" isn't doing anything to the stats cards here because that is there is only one div as a child of this div, but if you made each of these stats cards children of this larger grid div, then it would behave as you likely intended it to

Reply:
Sorry, actually there is a component called `StatsCard.tsx` that you should use instead of these.